publish-image/action.yml

```diff
@@ -110,8 +110,13 @@ runs:
         DIGEST=${IMAGE_REPO_DIGEST#*@}
 
         # Construct the package url (purl)
-        # TODO (@Techassi): Can we use 'oci' instead of 'docker' as the type?
-        PURL="pkg:docker/$IMAGE_REPOSITORY@$DIGEST?repository_url=$REGISTRY_URI"
+        URLENCODED_DIGEST=$(echo "$DIGEST" | sed 's/:/%3A/g')
+        URLENCODED_IMAGE_REPOSITORY=$(echo "$IMAGE_REPOSITORY" | sed 's/\//%2F/g')
```

Comment:
note: Are we sure that this covers all possible inputs and ensures the value is url-encoded correctly? We could instead use

```shell
$(echo "$REPO_DIGEST_OF_IMAGE" | jq -Rr @uri)
```

_Copied from https://github.com/stackabletech/operator-templating/pull/448#discussion_r1808169443_

Reply:
suggestion: Move the comment ("Construct ...") to where we actually construct the PURL. Add a comment here stating that we url-escape a bunch of values.

Reply:
Both done. I use `jq` to urlencode both values (because it would be confusing to use `jq` for one and `sed` for the other without a specific reason), so it's different from operator-templating now, but it should not make a difference in practice. It will also become irrelevant once we switch operator-templating to use this action.